### ISSUE TITLE
[DOCS] The salt(7) man page is one order of magnitude larger than any man page on the planet

### ISSUE THREAD

**Issue by sebastiancarlos** [saltstack/salt#64255]
**Description**
Dear SaltStack team, I hope this tongue-in-cheek issue finds you well.

I'm not going to beat around the bush. Look at this:

![image](https://user-images.githubusercontent.com/88276600/236840846-a6364d62-56c6-451c-b1fb-20effe6ffebe.png)
[(source)](https://gist.github.com/sebastiancarlos/6224faf860e37ebe467e7a58c6b58673)

My intentions when opening this issue is two-fold:
1. To offer my congratulations.
2. To bring this to your attention. There are not many conventions on how to structure man pages, and indeed there is some advantage in dumping all of one's documentation into a man page for easier global search. Still, I do wonder if we, as Homo Sapiens, have gone too far on this one.

**Suggested Fix**
I do not know.

This issue is for you to ponder, for no one has done what you did, so I consider you the only qualified party on the planet to decide how to deal with this gargantuan leviathan, which is roughly a fifth of the Encyclopædia Britannica.

**Additional context**
Have a lovely day

**Comment by barbaricyawps** [comment]
@sebastiancarlos , thanks for pointing this out.

In some of our unpublished training materials, I noticed this line:

> Salt ships with a number of man pages. The default man pages are standard descriptions of how to use the individual commands that ship with Salt, but a very large man page is included:

> `man 7 salt`

So, it looks like man 7 is the "everything but the kitchen sink" documentation. This is confirmed by a team member who told me:

> I think those pages are all of the Sphinx documentation, converted into Man pages. Salt docs, in PDF form, I think were 5000+ pages, so this is likely mapped

According to the unpublished training I referenced earlier, the preferred method of accessing the CLI documentation is to use the `status` command. For example, to view documentation for the status execution module:

`salt \*master sys.doc status`

You can also view the entire list of available functions:

`salt \*master sys.list_functions`

You can also list all functions for a specific module:

> salt \*master sys.list_functions cp

***

**TLDR;**

My take on this issue is that it sounds like an interesting project to evaluate our man page effectiveness and do some research into best practices and how to trim it down. That being said, it sounds like a time-intensive project that doesn't offer clear value to our users. For that reason, I'll keep this issue open and invite anyone in the community to work on it, they can certainly do so.

I welcome other people's takes on it, though.

**Comment by mthibaut** [comment]
Your mission, Jim, should you decide to accept it, is to find meaning in salt.7 and trim it down to below the number of lines in the bible, i.e. by a factor of roughly 20.

**Comment by ITJamie** [comment]
Worth bringing this up here for visibility. The salt-extensions migration could solve this but https://github.com/saltstack/salt/discussions/66144 would propose keeping all of the docs in salt core for extensions that get moved out

**Comment by Thaodan** [comment]
Would it make more sense to split the manpages up into different subpages? Projects like Git are a good example for this.
Further Sphinx can generate to various formats that the user might want to use on their machine such as Devhelp, html or GNU Info.

**Comment by twangboy** [comment]
This was fixed in 3006.11

**Comment by Thaodan** [comment]
> This was fixed in 3006.11

How was it fixed? I just see a removal, there isn't a replacement for the manpage.

**Comment by dwoz** [comment]
> > This was fixed in 3006.11
> 
> How was it fixed? I just see a removal, there isn't a replacement for the manpage.

We have no intention of trying to maintain a salt.7 man page moving forward. If this is something you'd like to do and can commit to actually maintaining it we're open to discussion.

**Comment by Thaodan** [comment]
I'm not asking to continue maintain the salt.7 man page but to have a
replacement similar to for example the git or systemd man pages were
each topic is split into a package like git-<topic> or systemd.<topic>.
